doc/macro_utils_design.md

```diff
@@ -0,0 +1,240 @@
+macro utils design
+=======
+
+Macro utils is a set of C useful macros.
+
+This document explains how it was build and what every macro does. IN supporting understand rather than code accuracy, some macros will be chopped to deal with a reasonable number of arguments rather than their implementation that usually supports above 120 arguments.
```

Comment:
>understand [](start = 81, length = 10)

understandability (or understanding the code)

Reply:
added ability

---
In reply to: [464137916](https://github.com/Azure/azure-macro-utils-c/pull/42#discussion_r464137916) [](ancestors = 464137916)